### PR TITLE
Loki: Convert Loki query response to dataframes

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
Change query response from time series directly to dataframes. To test, set up a panel with Alerts with a gdev-loki data source. Compare results on this branch and master. This PR is very similar to https://github.com/grafana/grafana/pull/31948. 

**Problem/Probably not related to this PR:**
The `formatLegend` works correctly and  when we have a legend, the name is created correctly. It is also correct when creating DataFrame. However, when we create Alerts using `dataFrames` and not `timeSeries`, the resulting name of Alert is incorrect. as you can see below, instead of the correct name and value `app=4.165151515151513` we have `value {cluster=ops-tools1, container=app, job=tns-demo/app, name=app, namespace=tns-demo, pod=app-8498cb7b9d-l7h4z, pod_template_hash=8498cb7b9d, stream=stdout}=1.8060606060606061`. (the actual values  are correct, just the name is incorrect). I don't think it is related to this PR, but wanted to bring this up. I have checked and it is also the case in Prometheus. Should I create separate issue for this? 
**Correct:**
image
**Incorrect:**
image

### CODE REVIEW

**Comment by dsotirakis** [comment]
Great work Ivana! 🎉  I think the thing you mention about the name formatting, has to do with the value displaying the tags (by default?). Let me have a quick look.

**Review by dsotirakis — APPROVED** [review]
LGTM! 

I think you are right though @ivanahuckova, this issue you found seems like a Frontend Issue to me, since the response on the backend side seems to have the appropriate format and stuff. I'll push a fix for prometheus and I'll post it here.

**Comment by dsotirakis** [comment]
@ivanahuckova FYI https://github.com/grafana/grafana/pull/32356

File: pkg/tsdb/loki/loki.go [hunk 1]
```diff
@@ -146,32 +146,34 @@ func (e *LokiExecutor) parseQuery(dsInfo *models.DataSource, queryContext plugin
...
-				null.FloatFrom(float64(k.Value)), null.FloatFrom(float64(k.Timestamp.Unix() * 1000)),
-			})
+			timeVector = append(timeVector, time.Unix(k.Timestamp.Unix(), 0).UTC())
+			values = append(values, float64(k.Value))
 		}
 
-		queryRes.Series = append(queryRes.Series, series)
+		frames = append(frames, data.NewFrame(name,
+			data.NewField("time", nil, timeVector),
+			data.NewField("value", tags, values)))
```
Comment (dsotirakis) [inline pkg/tsdb/loki/loki.go:174]:
```suggestion
		    data.NewField("value", tags, values).SetConfig(&data.FieldConfig{DisplayNameFromDS: name})))
```

Comment (dsotirakis) [reply pkg/tsdb/loki/loki.go:174]: See https://github.com/grafana/grafana/pull/32356

**Review by dsotirakis — APPROVED** [review]
Great @ivanahuckova! If you've already tested it, feel free to merge ✅

**Comment by ivanahuckova** [comment]
Yes, I have tested this and alerts have correct interpolated names. Thanks for reviewing. 🙏 
image

**Pull request merged**